http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -184,7 +186,16 @@ void closeChildChannels() {
      * Delivers the exception to all registered child channels, and prohibits new streams being created on this connection.
      */
     void closeChildChannels(Throwable t) {
-        closeAndExecuteOnChildChannels(ch -> ch.pipeline().fireExceptionCaught(t));
+        closeAndExecuteOnChildChannels(ch -> ch.pipeline().fireExceptionCaught(decorateConnectionException(t)));
+    }
+
+    private Throwable decorateConnectionException(Throwable t) {
+        String message = "An error occurred on the connection: " + t.getMessage();
+        if (IOException.class.isAssignableFrom(t.getClass())) {
```

Comment:
minor: `instanceof` here is more concise

Reply:
Fixed